crates/storage/nippy-jar/src/phf/fmph.rs

```diff
@@ -0,0 +1,111 @@
+use super::KeySet;
+use crate::NippyJarError;
+use ph::fmph::{BuildConf, Function};
+use serde::{
+    de::Error as DeSerdeError, ser::Error as SerdeError, Deserialize, Deserializer, Serialize,
+    Serializer,
+};
+use std::{clone::Clone, hash::Hash, marker::Sync};
+
+/// Wrapper struct for [`Function`]. Implementation of the following [paper](https://dl.acm.org/doi/10.1145/3596453).
+#[derive(Default)]
+pub struct Fmph {
+    function: Option<Function>,
+}
+
+impl Fmph {
+    pub fn new() -> Self {
+        Self { function: None }
+    }
+}
+
+impl KeySet for Fmph {
+    fn set_keys<T: AsRef<[u8]> + Sync + Clone + Hash>(
+        &mut self,
+        keys: &[T],
+    ) -> Result<(), NippyJarError> {
+        self.function = Some(Function::from_slice_with_conf(
+            keys,
+            BuildConf { use_multiple_threads: true, ..Default::default() },
+        ));
+        Ok(())
+    }
+
+    fn get_index(&self, key: &[u8]) -> Result<Option<u64>, NippyJarError> {
+        if let Some(f) = &self.function {
+            return Ok(f.get(key))
+        }
+        Err(NippyJarError::PHFMissingKeys)
+    }
+}
+
+impl PartialEq for Fmph {
+    fn eq(&self, other: &Self) -> bool {
+        match (&self.function, &other.function) {
+            (Some(func1), Some(func2)) => {
+                func1.level_sizes() == func2.level_sizes() &&
+                    func1.write_bytes() == func2.write_bytes() &&
+                    {
+                        #[cfg(not(test))]
```

Comment:
I would rather put the whole `impl PartialEq for Fmph` under `#[cfg(test)]` to prevent the accidental usages in non-test code